requirements/work-machine-client.tex

```diff
@@ -0,0 +1,20 @@
+\subsection{Work machine client(WMC)}
+\subsubsection{Main features}
+\begin{enumerate}
+  \item Each work machine shall have a single instance of WMC running.
+  \item On startup, WMC shall connect to the server and report its hardware capabilities. (TODO: define hw capabilities in glossary?)
+  \item WMC must be able to receive commands from the central server.
+  \item WMC shall support the following commands:
+  \begin{enumerate}
+    \item Start/Resume execution of a job.
+    \item Pause execution of a job.
+    \item Remove a job from the list of running jobs on the current machine.
```

Comment:
\item Query the current state of the work machine: running jobs, load on hardware.
\item Terminate this instance of WMC.

Reply:
> \item Query the current state of the work machine: running jobs, load on hardware.

Would it make sense if the WMC periodically sends notifications to the server about its load?
For example, if we want to show a website with current load of each work machine.

Reply:
I think the server should periodically ping the workers and request their current state.
That way we can also detect if there are network problems.

But now that I think about it, we are already talking design here.
And more broadly, aren't details about the interface of the worker client also design?
The user will never directly interact with a worker client, so should we already determine the commands that the worker client must support?

Reply:
> The user will never directly interact with a worker client, so should we already determine the commands that the worker client must support?

Yes, we don't need them if we do only a single high-level requirements document. But if we do lower-level requirement, then we should include such commands (this PR is about lower-level requirements, I should have stated this in the beginning).

This still leaves plenty of things to do in the design phase - what classes do we have in the worker? What methods? How do classes interact with each other?